_posts/2021-09-17-the-joy-of-slow-programming.md

```diff
@@ -0,0 +1,157 @@
+authentication it was just easier to access the JSON API directly. My
+abstraction was lightweight so I didn't have to modify it. I just wrote this
+one particular scraper differently.
+
+At some point I introduced the Apartment model to the application. At this
+stage I wasn't yet sure what data I can actually get in a consistent way. The
+only thing that I knew for sure was that I need to store a unique identifier for
+each apartment to avoid sending duplicated notifications after each subsequent
+check. That's why I added a mandatory `external_id` field. Everything else ended
+up in a JSONB `properties` field. This was simple and allowed me to move fast
+with delivering business value while moving slow with technical assumptions.
+
+### Starting to use the app myself
+
+A month or two have passed and guess what? I found myself in a need for a new
+apartment. Yes, I had to move out from my flat share and find a new apartment
+for myself. The application that I originally developed for my friend became my
+own hope on the cruel Berlin flat market.
+
+I was particularly interested in scoring an apartment in one of the housing
+associations (Wohnungsgenossenschaften), because they provide stable and secure
+renting possibilities and I believe in this organisational form. The domain
+situation here was different. The offers didn't appear often, sometimes you
+could wait for the whole month for one housing association to publish something.
+However, there were about 30-40 housing associations that I was interested in.
+
+I started with a long list of links and every day was going through all of them,
+mostly to find out that no new offers appeared. This was a tedious and
+frustrating process. So I began migrating the sources to be automatically handled
+by my app, one by one. From the technical point of view they were no different
+than the previous sources, so I didn't have to change anything in my assumptions.
+Again, I was working step by step, no more than one hour per day. Each migrated
+source meant one link less to click every morning, the list was shrinking, and
+the progress was visible to me. This kept me motivated.
+
+### Happy news
+
+One day I received a notification from a housing cooperative that was
+particularly appealing to me. The apartment was in a neighbourhood close to my
+old apartment and a really good deal for the money. I immediately applied for
+it and from that moment on things moved fast. Next day I had a viewing and
+within a week I signed the contract.
+
+The application that I developed as a side project scored me a new apartment!
+That's my most successful side project so far.
+
+I attribute this success to the philosophy of "slow programming" that I followed:
+1. put a hard limit on the time per day that you invest into the side project
+2. move in small steps, don't rush, and appreciate the little learnings that
+come every day
+3. stay motivated by making a tiny but visible improvement with each step
```

Comment:
I feel it would be nice to add a final thing like "Happy (slow-)coding!" or sth. like that, just to round it off :)

Reply:
Great idea!